docs/maven-plugin.md

```diff
@@ -0,0 +1,174 @@
+
+**static-code-analysis:report**
+
+Description:
+    Transforms the results from FindBugs, Checkstyle and PMD into a single HTML Report with XSLT
+
+Parameters:
+
+| Name | Type| Description |
+| ------ | ------| -------- |
+| **report.targetDir** | String | The directory where the individual report will be generated (default value is **${project.build.directory}/code-analysis**) |
+| **report.summary.targetDir** | String | The directory where the summary report, containing links to the individual reports will be generated (Default value is **${session.executionRootDirectory}/target**)|
+| **report.fail.on.error** | Boolean | Describes of the build should fail if high priority error is found (Default value is **true**)|
+
+## Customization
+
+Different sets of checks can be executed on different types of projects.
+
+The tool executes different checks on OSGi bundles and ESH Bindings. It uses default configuration files for FindBugs, Checkstyle and PMD that are stored in the `src/main/resources/configuration`.
+
+If you want to use a custom set of rules you will have to set the configuration parameters for the individual MOJOs. An example configuration may look like this;
+
+```
+  <plugin>
+    <groupId>org.openhab.tools</groupId>
+    <artifactId>static-code-analysis</artifactId>
+    <configuration>
+      <checkstyleRuleset>build-tools/checkstyle/binding.xml</checkstyleRuleset>
+      <checkstyleFilter>build-tools/checkstyle/suppressions.xml</checkstyleFilter>
+      <pmdRuleset>build-tools/pmd/binding.xml</pmdRuleset>
+      <findbugsInclude>build-tools/findbugs/binding.xml</findbugsInclude>
+      <findbugsExclude>build-tools/findbugs/exclude.xml</findbugsExclude>
+      <findbugsRuleset>build-tools/findbugs/visitors.xml</findbugsRuleset>
+    </configuration>
+  </plugin>
+```
+
+Information about the syntax of the configuration files (except the `visitors.xml`) can be found on the web pages of the individual plugins.
+
+The `visitors.xml` contains a list with FindBugs visitors (bug detectors) and has the following syntax:
+
+```
+<?xml version="1.0" encoding="UTF-8"?>
+<visitors>
+  <visitor>AtomicityProblem</visitor>
+  ...
+<visitors/>
+```
+
+### Individual plugin customization
+
+Each of the Maven plugins that are used (for FindBugs, Checkstyle and PMD) are configured by setting user properties that are located in the `src/main/resources/configuration` directory.
+
+You can refer to the following links for more configuration options for the specific Maven plugins:
+
+- https://maven.apache.org/plugins/maven-pmd-plugin/check-mojo.html;
+- https://maven.apache.org/plugins-archives/maven-checkstyle-plugin-2.16/checkstyle-mojo.html;
+- http://gleclaire.github.io/findbugs-maven-plugin/check-mojo.html.
+
+## Reuse Checks
+
+PMD, Checkstyle and FindBugs come with a set of custom rules that can be used directly in a rule set.
+
+Helpful resources with lists of the available checks and information how to use them:
+

```

Comment:
No, I think it doesn't make sense, here I have added some links to the checks that are part of PMD, Checkstyle or Findbugs.